spring-pulsar/src/main/java/org/springframework/pulsar/config/reactive/AbstractReactivePulsarListenerEndpoint.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.pulsar.config.reactive;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+import org.apache.commons.logging.LogFactory;
+import org.apache.pulsar.client.api.SubscriptionType;
+import org.apache.pulsar.common.schema.SchemaType;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.BeanFactoryAware;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.config.BeanExpressionContext;
+import org.springframework.beans.factory.config.BeanExpressionResolver;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.context.expression.BeanFactoryResolver;
+import org.springframework.core.log.LogAccessor;
+import org.springframework.expression.BeanResolver;
+import org.springframework.lang.Nullable;
+import org.springframework.pulsar.listener.adapter.PulsarMessagingMessageListenerAdapter;
+import org.springframework.pulsar.listener.reactive.ReactivePulsarMessageHandler;
+import org.springframework.pulsar.listener.reactive.ReactivePulsarMessageListenerContainer;
+import org.springframework.pulsar.support.MessageConverter;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+/**
+ * Base implementation for {@link ReactivePulsarListenerEndpoint}.
+ *
+ * @param <T> Message payload type.
+ * @author Christophe Bornet
+ */
+public abstract class AbstractReactivePulsarListenerEndpoint<T>
+		implements ReactivePulsarListenerEndpoint<T>, BeanFactoryAware, InitializingBean {
+
+	private final LogAccessor logger = new LogAccessor(LogFactory.getLog(getClass()));
```

Comment:
2 things:

* I don't think this is being used currently 
* there is also a `LogAccessor.<init>(Class)` which simplifies this a bit

Reply:
Done